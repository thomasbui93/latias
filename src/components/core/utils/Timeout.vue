<template>
  <span>{{ timeLeft }}</span>
</template>
<script>
export default {
  props: {
    timeout: {
      required: true,
      type: Number,
    },
  },
  data() {
    return {
      timeLeft: this.timeout,
    };
  },
  mounted() {
    const self = this;
    const countdown = setInterval(() => {
      if (self.timeLeft > 0) {
        self.timeLeft -= 1;
      } else {
        clearInterval(countdown);
      }
    }, 1000);
  },
};
</script>
