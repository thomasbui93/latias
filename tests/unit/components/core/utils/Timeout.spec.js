import { shallowMount } from '@vue/test-utils';
import Timeout from '@/components/core/utils/Timeout.vue';

describe('Timeout component', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(Timeout, {
      propsData: {
        timeout: 2000,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('set text for time remaining in the clock', () => {
    const wrapper = shallowMount(Timeout, {
      propsData: {
        timeout: 2,
      },
    });
    jest.advanceTimersByTime(1000);
    expect(wrapper.vm.timeLeft).toBe(1);
    jest.advanceTimersByTime(1000);
    expect(wrapper.vm.timeLeft).toBe(0);
    jest.advanceTimersByTime(1000);
    expect(wrapper.vm.timeLeft).toBe(0);
  });
});
