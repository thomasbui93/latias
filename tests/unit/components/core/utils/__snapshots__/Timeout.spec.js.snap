// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeout component matches snapshot 1`] = `<span>2000</span>`;
